### PR TITLE
WFLY-13887 Add http-connections element to jboss-ejb-client_1_4.xsd s…

### PR DESCRIPTION
**Pull request by sudeshnas93**
Issue: https://issues.redhat.com/browse/WFLY-13887

### CODE REVIEW

File: docs/src/main/asciidoc/_client-guide/jboss-ejb-client.adoc [hunk 1]
```diff
@@ -50,6 +53,12 @@ client context, should use pass-by-reference (instead of pass-by-value) semantic
 * `outbound-connection-ref` Reference to an outbound connection configured in the remoting subsystem.
 * `connect-timeout` The timeout, in milliseconds, to be used while creating a connection.
 
+[[http-connections]]
+=== <http-connections>
+
+* `http-connection` HTTP Connection for EJB invocation.
```
Comment (chengfang) [inline docs/src/main/asciidoc/_client-guide/jboss-ejb-client.adoc:59]: `http-connection` should be documented as a sub-element in its subsection under `http-connections`, just like `remoting-ejb-receiver` is a subelement of `ejb-receivers`.

File: ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd [hunk 2]
```diff
@@ -162,6 +162,37 @@
...
+    </xsd:choice>
+        <xsd:attribute name="uri" type="xsd:string" use="required">
+            <xsd:annotation>
+                <xsd:documentation>
+                    Uniform Resource Identifier for the HTTP connection
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:attribute>
+    </xsd:complexType>
+
```
Comment (chengfang) [inline ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd:195]: the indentation is a bit off. Need to reformat the addition.

File: ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd [hunk 3]
```diff
@@ -162,6 +162,37 @@
         </xsd:attribute>
     </xsd:complexType>
 
+    <xsd:complexType name="http-connectionsType">
+        <xsd:choice>
+        <xsd:element name="http-connections" type="http-connectionsType" >
```
Comment (chengfang) [inline ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd:167]: this should be `http-connection`

File: ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd [hunk 4]
```diff
@@ -162,6 +162,37 @@
...
+            <xsd:annotation>
+                <xsd:documentation>
+                    Configures remote http-connections for EJB invocation
```
Comment (chengfang) [inline ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd:170]: should be http-connection

File: ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd [hunk 5]
```diff
@@ -162,6 +162,37 @@
         </xsd:attribute>
     </xsd:complexType>
 
+    <xsd:complexType name="http-connectionsType">
+        <xsd:choice>
```
Comment (chengfang) [inline ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd:166]: Need min and max bound for choice:
        <xsd:choice minOccurs="0" maxOccurs="unbounded">

File: ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd [hunk 6]
```diff
@@ -162,6 +162,37 @@
...
+    <xsd:complexType name="http-connectionType">
+    <xsd:choice>
+        <xsd:element name="http-connection" type="http-connectionType" >
+            <xsd:annotation>
+                <xsd:documentation>
+                    Configures remote HTTP Connection for EJB invocation
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:element>
+    </xsd:choice>
```
Comment (chengfang) [inline ejb3/src/main/resources/schema/jboss-ejb-client_1_4.xsd:186]: Need to remove the above `choice` element.

**Comment by chengfang** [comment]
In the xsd file, we need to:
* add the new http-connections element to client-contextType
* add a new element type http-connectionsType
* add a new element type http-connectionType

It should follow the same pattern as the element ejb-receivers, which is also a sub-element of client-context.

**Comment by bstansberry** [comment]
@sudeshnas93 @chengfang We should get this xsd correct for WF 21 as it's a new version.

**Comment by sudeshnas93** [comment]
@chengfang Fixed accordingly.

**Review by chengfang — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @sudeshnas93 !

**Pull request merged**